src/__tests__/waitFor.test.tsx

```diff
@@ -204,6 +204,37 @@ test.each([false, true])(
   }
 );
 
+const fibonaci = (n: number): number => {
+  if (n === 0 || n === 1) {
+    return 1;
+  }
+
+  return fibonaci(n - 1) + fibonaci(n - 2);
+};
```

Comment:
Could we replace this with something both more controllable and explicit:
```suggestion
function blockThread(milliseconds)
{
    var end = Date.now() + milliseconds;
    while (Date.now() < end) {
      // Intentionally do nothing
    };
}
```

Reply:
Very good idea, it's way more explicit that way! I just had to switch to real timers when calling blockThread so that it works